WIP: 12.0 app connection public profile

njeudy opened the pull request:
Initial work on profile new data model for numeric currency:

- Add res.partner.backend model (not res.partner.bank !)
- use only public_profile for data in API

TODO:
- test
- Sync there is a mistake with partner_id value returned and need more work.
- Add res.partner.backend tree, form, view et filter field visibility on "type" value (comchain or cyclos)

Commit:
[NEW] add res.partner.backend model and link to res.partner

Commit:
[NEW] use res.partner.backend for comchain services

Commit:
[NEW] rename _comchain_backend_data to _comchain_backend_json_data

Commit:
[NEW] use res.partner.backend for cyclos service

Commit:
[NEW] use public_profile_id data for all api response

stephansainleger commented on lcc_comchain_base/models/res_partner.py:
```diff
@@ -80,7 +120,7 @@ def _comchain_backend_data(self):
                     "wallet": wallet,
-                    "message_key": self.comchain_message_key,
+                    "message_key": backend_data.comchain_message_key,
                     "active": self.comchain_active,
```
Remove call to old variables:
`backend_data.status == 'active'` instead of `self.comchain_active`

stephansainleger replied:
Done

stephansainleger commented on lcc_cyclos_base/models/res_partner.py:
```diff
@@ -278,30 +305,32 @@ def addCyclosUser(self):
-                        "cyclos_active": True,
                         "cyclos_status": data.get("status", ""),
+                        "status": "active",
                     }
```
Remove call to old variable:
- remove `record.cyclos_create_response = res.text`
- add `"cyclos_create_response" : res.text,` in the backend_date.write call below

stephansainleger replied:
Done

stephansainleger commented on lcc_cyclos_base/models/res_partner.py:
```diff
@@ -264,12 +288,15 @@ def addCyclosUser(self):
                         "cyclos_status": data.get("status", ""),
+                        "name": "cyclos:%s" % data.get("user")["id"],
+                        "status": "to_confirm",
                     }
```
Remove call to old variable:
-  remove `record.cyclos_create_response = res.text` above
-  add `"cyclos_create_response"` : res.text, in the `backend_obj.create` call

stephansainleger replied:
Done

stephansainleger commented on lcc_cyclos_base/models/res_partner.py:
```diff
@@ -105,29 +126,30 @@ def _update_auth_data(self, password):
-        backend_data = self._cyclos_backend_data()
-        if self._cyclos_backend_id and backend_data and self.cyclos_active:
+        backend_json_data = self._cyclos_backend_json_data()
+        if self._cyclos_backend_id and backend_json_data and self.cyclos_active:
```
Remove call to old variable: replace `self.cyclos_active` by `backend_json_data.active`

stephansainleger replied:
Done

stephansainleger commented on lcc_cyclos_base/models/res_partner.py:
```diff
@@ -82,21 +93,31 @@ def _cyclos_backend_id(self):
+                    "owner_id": backend_data.cyclos_id,
                     "url": self.env.user.company_id.cyclos_server_url,
-                    "active": self.cyclos_active,
+                    "active": backend_data.cyclos_active,
```
`cyclos_active` doesn't exist in res.partner.backend. To replace with `"active": backend_data.status == 'active'` ?

stephansainleger replied:
Done

stephansainleger commented:
Other issue spotted in the code, but I couldn't link a comment to the line because it was an unchanged line : 
#170 of file l`cc_cyclos_base/models/res_partner.py` : `if self.cyclos_id` is a call to an old variable, which must be changed to a call to a res.partner.backend object

Commit:
[UPD] Modification of cyclos message management

to support other kind of messages like "generalErrors"

Commit:
[NEW] Add lcc_backend form view

Commit:
[NEW] Add inherited lcc_backend form view for Cyclos

Commit:
[UPD] Merge open_commercial_member_entity at parent addon level

Commit:
[NEW] Add inherited lcc_backend form view for comchain

Commit:
[UPD] Code refactoring between res.partner and res.partner.backend

Move of several methods in res.partner.backend model

Commit:
[NEW] Add Cyclos backend management buttons

Commit:
[UPD] Remove automatic validation of cyclos wallet for persons

because this automatic process can fail if the Cyclos is configured to
automatically validate the new users.

Commit:
[UPD] Improvement of cyclos status management

Status field is now computed, and status manipulation is possible only
if conditions validated.

Commit:
[FIX] Correction on security rule to allow search on partners

Signed-off-by: Valentin Lab <valentin.lab@kalysto.org>

Commit:
[FIX] make search api entry point working again

- Update for new members management (profile)
- Clean search domain

Signed-off-by: Valentin Lab <valentin.lab@kalysto.org>

Commit:
[IMP] clean unused api entry points

vaab committed:
[CLEAN] space related change (cosmetics)

Signed-off-by: Valentin Lab <valentin.lab@kalysto.org>
